plutus-tx-plugin/test/Strictness/Spec.hs

```diff
@@ -0,0 +1,109 @@
+{-# LANGUAGE BangPatterns        #-}
+{-# LANGUAGE DataKinds           #-}
+{-# LANGUAGE OverloadedStrings   #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE Strict              #-}
+{-# LANGUAGE TemplateHaskell     #-}
+{-# OPTIONS_GHC -fplugin PlutusTx.Plugin #-}
+
+module Strictness.Spec where
+
+import Test.Tasty.Extras
+
+import PlutusTx qualified as Tx
+import PlutusTx.Code
+import PlutusTx.Prelude qualified as PlutusTx
+import PlutusTx.Test
+import PlutusTx.TH (compile)
+
+tests :: TestNested
+tests =
+  testNested
+    "Strictness"
+    [ goldenEvalCekCatch "lambda-default" $ [lambdaDefault `unsafeApplyCode` bot]
+    , goldenPirReadable "lambda-default" lambdaDefault
+    , goldenUPlcReadable "lambda-default" lambdaDefault
+
+    -- FIXME: This should not crash, but it currently does.
+    , goldenEvalCekCatch "lambda-nonstrict" $ [lambdaNonStrict `unsafeApplyCode` bot]
+    , goldenPirReadable "lambda-nonstrict" lambdaNonStrict
+    , goldenUPlcReadable "lambda-nonstrict" lambdaNonStrict
+
+    , goldenEvalCekCatch "lambda-strict" $ [lambdaStrict `unsafeApplyCode` bot]
+    , goldenPirReadable "lambda-strict" lambdaStrict
+    , goldenUPlcReadable "lambda-strict" lambdaStrict
+
+    -- FIXME: This should crash (because the `Strict` extension is on), but it currently doesn't.
+    , goldenEvalCekCatch "let-default" $ [letDefault `unsafeApplyCode` one]
+    , goldenPirReadable "let-default" letDefault
```

Comment:
Why not pretty-print the same program that you're testing? Would be easier to review and generally make sense of.

Reply:
I could do that but printing the argument doesn't really add anything. One should only ever need to check what the function is compiled into.